src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs

```diff
@@ -0,0 +1,294 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.Extensions.Logging;
+using Microsoft.ServiceFabric.Data;
+using Microsoft.ServiceFabric.Data.Collections;
+
+namespace ReleasePipelineRunner
+{
+    /// <summary>
+    /// An instance of this class is created for each service replica by the Service Fabric runtime.
+    /// </summary>
+    public sealed class ReleasePipelineRunner : IServiceImplementation, IReleasePipelineRunner
+    {
+        public ReleasePipelineRunner(
+            IReliableStateManager stateManager,
+            ILogger<ReleasePipelineRunner> logger,
+            BuildAssetRegistryContext context)
+        {
+            StateManager = stateManager;
+            Logger = logger;
+            Context = context;
+        }
+
+        public IReliableStateManager StateManager { get; }
+        public ILogger<ReleasePipelineRunner> Logger { get; }
+        public BuildAssetRegistryContext Context { get; }
+
+        private const string RunningPipelineDictionaryName = "runningPipelines";
+        private static HashSet<string> InProgressStatuses = new HashSet<string>(StringComparer.OrdinalIgnoreCase) { "notstarted", "scheduled", "queued", "inprogress" };
+
+        public async Task StartAssociatedReleasePipelinesAsync(int buildId, int channelId)
+        {
+            IReliableConcurrentQueue<ReleasePipelineRunnerItem> queue =
+                await StateManager.GetOrAddAsync<IReliableConcurrentQueue<ReleasePipelineRunnerItem>>("queue");
+            using (ITransaction tx = StateManager.CreateTransaction())
+            {
+                await queue.EnqueueAsync(
+                    tx,
+                    new ReleasePipelineRunnerItem
+                    {
+                        BuildId = buildId,
+                        ChannelId = channelId
+                    });
+                await tx.CommitAsync();
+            }
+        }
+
+        public async Task RunAsync(CancellationToken cancellationToken)
+        {
+            IReliableConcurrentQueue<ReleasePipelineRunnerItem> queue =
+                await StateManager.GetOrAddAsync<IReliableConcurrentQueue<ReleasePipelineRunnerItem>>("queue");
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                try
+                {
+                    using (ITransaction tx = StateManager.CreateTransaction())
+                    {
+                        ConditionalValue<ReleasePipelineRunnerItem> maybeItem = await queue.TryDequeueAsync(
+                            tx,
+                            cancellationToken);
+                        if (maybeItem.HasValue)
+                        {
+                            ReleasePipelineRunnerItem item = maybeItem.Value;
+                            using (Logger.BeginScope(
+                                "Triggering release pipelines associated with channel {channelId} for build {buildId}.",
```

Comment:
I think you're missing a $ prefix for string interpolation here.

Reply:
That's a good catch. We have the same bug in DependencyUpdater where I stole this code from. I'll update it there as well.

Reply:
Nevermind! synced with @alexperovich this is basically a string.Format

Reply:
This is actually supposed to not use string interpolation. This is using the Microsoft.Extensions.Logging logger with the string.Format arg style. The names in the {} become property names in the telemetry item.